python-sdk/nuscenes/map_expansion/arcline_path_utils.py

```diff
@@ -0,0 +1,200 @@
+# nuScenes dev-kit.
+# Code written by Freddy Boulton, 2020.
+
+import math
+from typing import Dict, Any, List, Tuple
+
+
+def principal_value(angle_in_radians: float) -> float:
+    """
+    Ensures the angle is within [-pi, pi).
+    :param angle_in_radians: Angle in radians.
+    :return: Scaled angle in radians.
+    """
+
+    interval_min = -math.pi
+    two_pi = 2 * math.pi
+    scaled_angle = (angle_in_radians - interval_min) % two_pi + interval_min
+    return scaled_angle
+
+
+def compute_segment_sign(arcline_path_3: Dict[str, Any]) -> Tuple[float, float, float]:
+    """
+    Compute the sign of an arcline_path_3 based on its shape.
+    :param arcline_path_3: arcline_patch_3 record.
+    :return: Tuple of signs for all three parts of the path. 0 if straight, -1 if right,
+        1 if left.
+    """
+    shape = arcline_path_3['shape']
+    segment_sign = [0, 0, 0]
+
+    if shape in ("LRL", "LSL", "LSR"):
+        segment_sign[0] = 1
+    else:
+        segment_sign[0] = -1
+
+    if shape == "RLR":
+        segment_sign[1] = 1
+    elif shape == "LRL":
+        segment_sign[1] = -1
+    else:
+        segment_sign[1] = 0
+
+    if shape in ("LRL", "LSL", "RSL"):
+        segment_sign[2] = 1
+    else:
+        segment_sign[2] = -1
+
+    return segment_sign
+
+
+def get_transformation_at_s(lie_group_pose: Tuple[float, float, float],
```

Comment:
@freddyaboulton Let's just call this `pose` instead of `lie_group_pose`.

Reply:
Can we also add an alias for `Pose = Tuple[float, float, float]` to simplify the type signatures?

Reply:
Done!